docs/content/docs/getting-started/build-transaction.mdx

```diff
@@ -315,32 +321,32 @@ const rpcSubscriptions = {} as RpcSubscriptions<SolanaRpcSubscriptionsApi>;
 const wallet = {} as TransactionSigner & MessageSigner;
 // ---cut-before---
 import {
-    getComputeUnitEstimateForTransactionMessageFactory, // [!code ++]
-    prependTransactionMessageInstruction, // [!code ++]
+    appendTransactionMessageInstruction, // [!code ++]
     // ...
 } from '@solana/kit';
-import { getSetComputeUnitLimitInstruction } from '@solana-program/compute-budget'; // [!code ++]
+import {
+    estimateComputeUnitLimitFactory, // [!code ++]
+    getSetComputeUnitLimitInstruction, // [!code ++]
+} from '@solana-program/compute-budget';
 
 let client: Client | undefined;
 export async function createClient(): Promise<Client> {
     if (!client) {
         // ...
 
         // Create a function to estimate and set the compute unit limit. // [!code ++]
-        const estimateComputeUnitLimit = getComputeUnitEstimateForTransactionMessageFactory({
-            rpc,
-        }); // [!code ++]
-        const estimateAndSetComputeUnitLimit = async <T extends CompilableTransactionMessage>(
+        const estimateComputeUnitLimit = estimateComputeUnitLimitFactory({ rpc });
+        const estimateAndSetComputeUnitLimit = async <
+            T extends BaseTransactionMessage & TransactionMessageWithFeePayer,
+        >(
             transactionMessage: T,
         ) => {
-            // [!code ++]
-            const computeUnitsEstimate = await estimateComputeUnitLimit(transactionMessage); // [!code ++]
-            return prependTransactionMessageInstruction(
-                // [!code ++]
-                getSetComputeUnitLimitInstruction({ units: computeUnitsEstimate }), // [!code ++]
-                transactionMessage, // [!code ++]
-            ); // [!code ++]
-        }; // [!code ++]
+            const computeUnitsEstimate = await estimateComputeUnitLimit(transactionMessage);
+            return appendTransactionMessageInstruction(
+                getSetComputeUnitLimitInstruction({ units: computeUnitsEstimate }),
+                transactionMessage,
+            ) as unknown as T;
```

Comment:
This `as unknown as T` is visible in the documentation which is 🤮 . 

A subsequent PR (https://github.com/anza-xyz/kit/pull/739) in this stack fixes this.